lib/tasks/fixity_check.rake

```diff
@@ -2,9 +2,75 @@ namespace :scholars_archive do
   desc "repository fixity check"
   task fixity: :environment do
     Rails.logger.warn "Running Hyrax::RepositoryFixityCheckService"
-    # OVERRIDE: From Hyrax, add async option for Fixity Check
+
+    # SETUP: Variables will be used in the case
+    failed_arr = []
+
+    # MAILER: Disable mailer so Fixity won't send out email while running check
+    ActionMailer::Base.perform_deliveries = false
+
+    # CREATE: Make a start time to know when Fixity starts
+    start_time = Time.now
+
+    # OVERRIDE: From Hyrax, add async option for Fixity check
     ::FileSet.find_each do |file_set|
       Hyrax::FileSetFixityCheckService.new(file_set, async_jobs: false).fixity_check
     end
+
+    # CREATE: Make an end time to know when Fixity finish
+    end_time = Time.now
+
+    # QUERY #1: Query data from the ChecksumAuditLog
+    latest_file = ChecksumAuditLog.where("updated_at >= ?", start_time)
+    file_checked = latest_file.count
+    file_passed = latest_file.where("passed = true").count
+    file_failed = latest_file.where("passed = false").count
+
+    # QUERY #2: Get all the ids that failed via checking with fixity
+    latest_file.each do |c|
+      if (c.passed != true)
+        failed_arr.append(c.file_set_id)
+      end
+    end
+
+    # APPEND: Display an outro message saying Fixity is done checking
+    Rails.logger.info "\n[COMPLETE] All fixity checks complete!\n"
+
+    # APPEND: Attach additional information to the log
+    Rails.logger.info "'START TIME': #{start_time.strftime("%B %-d, %Y %l:%M:%S:%L %p").to_s}"
+    Rails.logger.info "'END TIME': #{end_time.strftime("%B %-d, %Y %l:%M:%S:%L %p").to_s}"
+    Rails.logger.info "'No. of File Sets [CHECKED]': #{file_checked.to_s}"
+    Rails.logger.info "'No. of File Sets [PASSED]': #{file_passed.to_s}"
+    Rails.logger.info "'No. of File Sets [FAILED]': #{file_failed.to_s}\n"
+
+    Rails.logger.info "'No. of File Sets [IDs] that failed':"
+    if (failed_arr.empty?)
+      Rails.logger.info "None\n"
+    else
+      failed_arr.each do |f|
+        Rails.logger.info "#{f.to_s}"
```

Comment:
Ask @decimalator if he wants a failed ID per line, or a single summary array line.

Reply:
I think let's try a single summary array and see how that looks. 

as long as it's one line it can be as long as we want it to be. I think there's a limit to the length of a single log push to  Loki, I'll verify that. but the summary array shouldn't be any longer than some of the logs I've seen come through

Reply:
Hi @decimalator, below is two screenshots if we printed them out onto the log console via each `individual item` or in a `single summary array` format. Let me know if you would still like to continue with the `single summary array` style after checking them out below.

PS: This is all test data to display

`Option #1`: With `individual item`
<img width="690" alt="Screenshot 2022-12-14 at 10 24 40 AM" src="https://user-images.githubusercontent.com/111306182/207682816-2b158490-dd9c-4c9e-9063-a5139bfcbf96.png">

`Option #2`: With the `single summary array`
<img width="711" alt="Screenshot 2022-12-14 at 10 36 09 AM" src="https://user-images.githubusercontent.com/111306182/207683172-595e8c1e-3747-4064-aab0-89c5c40c0f03.png">

Reply:
I like option 2! looks great!

Reply:
@decimalator, thank you for the feedback, I will go with it. Also, another question I wanted to ask, do you want the single summary array be printed in the email format too or keep them original format right now which follow the resemblance of option 1?